src/main/java/school/hei/haapi/service/PaymentService.java

```diff
@@ -71,11 +72,24 @@ public List<Payment> getByStudentIdAndFeeId(
   public void computeRemainingAmount(String feeId, int amount) {
     Fee associatedFee = feeService.getById(feeId);
     associatedFee.setRemainingAmount(associatedFee.getRemainingAmount() - amount);
+    computeUserStatusAfterPayingFee(associatedFee.getStudent());
     if (associatedFee.getRemainingAmount() == 0) {
       associatedFee.setStatus(PAID);
     }
   }
 
+  public User computeUserStatusAfterPayingFee(User userToResetStatus) {
+    Instant now = Instant.now();
+    List<Fee> unpaidFeesBeforeNow =
+        feeRepository.getStudentPaidFeesFrom(now, userToResetStatus.getId(), LATE);
+    if (!unpaidFeesBeforeNow.isEmpty()) {
+      userToResetStatus.setStatus(User.Status.SUSPENDED);
+      return userRepository.save(userToResetStatus);
+    }
+    userToResetStatus.setStatus(User.Status.ENABLED);
+    return userRepository.save(userToResetStatus);
```

Comment:
again, updating the whole object is very risky, why not solely updating the status ?